php-css-parser-install-composer-plugin/src/Plugin.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * Plugin class.
+ *
+ * @package AmpProject\AmpWP\PhpCssParserInstall
+ */
+
+namespace AmpProject\AmpWP\PhpCssParserInstall;
+
+use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\IO\IOInterface;
+use Composer\Package\AliasPackage;
+use Composer\Package\CompleteAliasPackage;
+use Composer\Package\CompletePackageInterface;
+use Composer\Package\Link;
+use Composer\Package\Loader\ArrayLoader;
+use Composer\Package\Locker;
+use Composer\Plugin\PluginInterface;
+use Composer\Repository\InstalledRepositoryInterface;
+use Composer\Repository\LockArrayRepository;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use Composer\Semver\Constraint\Constraint;
+
+/**
+ * This Composer plugin patches the`sabberworm/php-css-parser` package so that it autoloads the correct namespace.
+ */
+class Plugin implements PluginInterface, EventSubscriberInterface {
+
+	/** @var string */
+	const CSS_PARSER_PACKAGE_NAME = 'sabberworm/php-css-parser';
+
+	/**
+	 * Apply plugin modifications to Composer
+	 *
+	 * @param Composer    $composer Composer instance.
+	 * @param IOInterface $io       IO instance.
+	 */
+	public function activate( Composer $composer, IOInterface $io ) {
+	}
+
+	/**
+	 * Remove any hooks from Composer
+	 *
+	 * This will be called when a plugin is deactivated before being
+	 * uninstalled, but also before it gets upgraded to a new version
+	 * so the old one can be deactivated and the new one activated.
+	 *
+	 * @param Composer    $composer Composer instance.
+	 * @param IOInterface $io       IO instance.
+	 */
+	public function deactivate( Composer $composer, IOInterface $io ) {
+	}
+
+	/**
+	 * Prepare the plugin to be uninstalled
+	 *
+	 * This will be called after deactivate.
+	 *
+	 * @param Composer    $composer Composer instance.
+	 * @param IOInterface $io       IO instance.
+	 */
+	public function uninstall( Composer $composer, IOInterface $io ) {
+	}
+
+	/** @inerhitDoc */
+	public static function getSubscribedEvents() {
+		return [
+			ScriptEvents::PRE_AUTOLOAD_DUMP => 'onPreAutoloadDump',
+		];
+	}
+
+	/**
+	 * Patch the package before the autoload file is generated.
+	 *
+	 * @param Event $event Composer event.
+	 */
+	public static function onPreAutoloadDump( Event $event ) {
+		$package_locker   = $event->getComposer()->getLocker();
+		$local_repository = $event->getComposer()->getRepositoryManager()->getLocalRepository();
+
+		self::patchComposerLockFile( $package_locker );
```

Comment:
There may be an event earlier than this that we could hook into so that we only would have to update the package instance once and have Composer generate the correct `composer.lock` but I didn't spend much time seeing if that was possible.